Library/src/Systems/Assets/AssetSystem.cpp

```diff
@@ -1068,14 +1074,22 @@ void AssetSystem::GetMaterials(const csp::common::String& SpaceId, GLTFMaterials
 					// Finish
 					GLTFMaterialsResult Result(DownloadResult.GetResultCode(), DownloadResult.GetHttpResultCode());
 					Result.SetGLTFMaterials(SpaceMaterials);
+
 					Callback(Result);
+					Finished = true;
 				}
 			};
 
 			for (size_t i = 0; i < Assets.Size(); ++i)
 			{
 				DownloadAssetData(Assets[i], DownloadMaterialCallback);
 			}
+
+			// We have to wait so this function doesn't exit and destroy "AssetsDownloaded"
+			while (!Finished)
+			{
+				std::this_thread::sleep_for(std::chrono::milliseconds {10});
+			}
```

Comment:
[fix] Don't busy wait, use a proper async primitive if you can.

Reply:
Updated to future/promise

Reply:
👍